### DISCUSSION TITLE
Getting "Type instantiation is excessively deep and possibly infinite" for non-recursive schema inference

### DISCUSSION THREAD

**Discussion by PiotrJander** [colinhacks/zod#1459] [Q&A]
I created this fairly complex (but non-recursive) Zod schema with discriminated unions: https://gist.github.com/PiotrJander/b72b9c4586f653b840f34ff749d171be

When I do type inference

```
export type ReportOrder = z.infer<typeof ReportOrderSchema>;
```

I get

```
error TS2589: Type instantiation is excessively deep and possibly infinite.
```

Are there any techniques to help TS compute this type? Breaking it down into sub-schemas? Passing type parameters?

Help will be appreciated, thanks!

**Comment by Sotatek-TungLe2** [comment]
Try to update your typescript to ^4.5.0

**Reply by PiotrJander** [reply]
Thanks, I'm already using TS 4.7. I think I'll try to infer the types for each member of the discriminated union individually, and then join them into a union on the type level.
